openstates/dc/bills.py

```diff
@@ -1,212 +1,449 @@
-
-        vote.add_source(url)
-
-        # members are only text on page in a <u> tag
-        for member_u in doc.xpath('//u'):
-            member = member_u.text
-            # normalize case
-            vote_text = member_u.xpath('../../i/text()')[0].upper()
-            if 'YES' in vote_text:
-                vote.yes(member)
-            elif 'NO' in vote_text:
-                vote.no(member)
+        for v in vote_info:
+            vote_count = 0 if v["VoteCount"] == "" else int(v["VoteCount"])
+
+            if v["VoteType"] == "Yes":
+                yes_count = vote_count
+            elif v["VoteType"] == "No":
+                no_count = vote_count
             else:
-                vote.other(member)
+                other_count += vote_count
+
+        passed = False
+        if yes_count > no_count:
+            passed = True
+
+        vote = Vote("upper",date,"Committee Vote",passed,yes_count,no_count,other_count)
         bill.add_vote(vote)
 
-    def scrape(self, session, chambers):
-        url = 'http://dcclims1.dccouncil.us/lims/print/list.aspx?FullPage=True&Period=' + session
-
-        html = self.urlopen(url)
-        doc = lxml.html.fromstring(html)
-        rows = doc.xpath('//table/tr')
-        # skip first row
-        for row in rows[1:]:
-            bill_id, title, _ = row.xpath('td/text()')
-            title = title.replace(u'\xe2\x80\x99', "'")  # smart apostrophe
-
-            if bill_id.startswith('B'):
-                type = 'bill'
-            elif bill_id.startswith('CER'):
-                type = 'resolution'
-            elif bill_id.startswith('CA'):
-                type = 'contract'
-            elif bill_id.startswith('PR'):
-                type = 'resolution'
-            # don't collect these
-            elif (bill_id.startswith('GBM') or
-                  bill_id.startswith('HFA') or
-                  bill_id.startswith('IG')):
-                continue
+
+    def add_documents(self,attachment_path,bill,is_version=False):
+        global bill_versions
+        base_url = "http://lims.dccouncil.us/Download/" #nothing is actual links. we'll have to concatenate to get doc paths (documents are hiding in thrice-stringified json. eek.)
+        for a in attachment_path:
+            doc_type = a["Type"]
+            doc_name = a["Name"]
+            rel_path = a["RelativePath"]
+            if doc_type and doc_name and rel_path:  
+                doc_url = base_url+rel_path+"/"+doc_name
             else:
-                # will break if type isn't known
-                raise ValueError('unknown bill type: %s' % bill_id)
+                self.logger.warning("Bad link for document {}".format(doc_name))
+                return
+
+            mimetype = "application/pdf" if doc_name.endswith("pdf") else None
+
+            #figure out if it's a version from type/name
+            possible_version_types = ["SignedAct","Introduction","Enrollment","Engrossment"]
```

Comment:
Nope! Just making sure it had the same name / was accounted for!

On Wed, Feb 25, 2015 at 1:54 PM, Rachel <notifications@github.com> wrote:

> In openstates/dc/bills.py
> <https://github.com/sunlightlabs/openstates/pull/588#discussion_r25367219>
> :
>
> >              else:
> > -                # will break if type isn't known
> > -                raise ValueError('unknown bill type: %s' % bill_id)
> > +                self.logger.warning("Bad link for document {}".format(doc_name))
> > +                return
> > +
> > +            mimetype = "application/pdf" if doc_name.endswith("pdf") else None
> > +
> > +            #figure out if it's a version from type/name
> > +            possible_version_types = ["SignedAct","Introduction","Enrollment","Engrossment"]
>
> I think that's enrollment. want me to call it "as amended"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sunlightlabs/openstates/pull/588/files#r25367219>.
>



-- 
Paul Tagliamonte
Software Developer | Sunlight Foundation